### PR TITLE
[PULL REQUEST] Implement a check for a minimum version (add add some function prototypes) -- Closes #31

### LINKED ISSUE #31: [FEATURE REQUEST] Add optional toggle to check if a mechanism needs a minimum KPP version
I was thinking it might be a nice feature to have the ability to add an optional KPP toggle such as `#MINIMUM_REQUIRED X.Y.Z` so that KPP will throw an error unless you are using version X.Y.Z or higher.

For example, the CMake build system lets you specify a minimum version such as:
```cmake
cmake_minimum_required (VERSION 3.13)
```

This might become important for models such as GEOS-Chem where we have many different versions in use.   For example, if someone wanted to use a mechanism with the new `#AUTOREDUCE` capability that @jimmielin and @msl3v are developing, you would need to use KPP 3.0.0 (when it is ready).   But if you don't want to use the #AUTOREDUCE feature then you wouldn't need to specify a minimum version.

I can work on adding this to the add-geos-chem-updates branch.

### PR DESCRIPTION
**Pull request by yantosca**
This PR corresponds to feature request #31.  We have now added:

  1. The `#MINVERSION` switch can now define a minimum version for a KPP mechanism.  If the KPP version is older than this minimum version, it will stop with a fatal error.  (If `#MINVERSION` is missing, then this error check will be skipped).
  2. Updated the C-I tests to add the `ros_minver` test... this is a Rosenbrock mechanism but it should fail on the minimum version error check.  The C-I test scripts will interpret its failure as a succes.
  3. Also added some missing function prototypes, in order to remove the -Wimplicit-function-declaration warning messages at compile time.

### CODE REVIEW

**Comment by yantosca** [comment]
I will attend to the failing C-I test soon.  I think there may be an issue in the script.

**Comment by yantosca** [comment]
Or actually the build test failed on #MINVERSION but the Docker container interpreted that as an error.  I will doublecheck the script logic.
```console
This is KPP-2.4.0.

KPP is parsing the equation file.

You are using KPP 2.4.0, but your mechanism needs version none or later.

Fatal error : 1 errors and 0 warnings encountered.
Program aborted
The command '/bin/sh -c /kpp/.ci-pipelines/ci-testing-script.sh' returned a non-zero code: 1
##[error]The command '/bin/sh -c /kpp/.ci-pipelines/ci-testing-script.sh' returned a non-zero code: 1
```

**Comment by yantosca** [comment]
I will also update the latex manual file in a subsequent commit.

**Comment by yantosca** [comment]
The CI test problem is now fixed in commit f1f06310be5937f8cf5798b854a352ade51ebfe8.  This was due to improper string comparison in the function `KppVersionIsTooOld`.  Using `strncmp` in the test fixed it.  Shout out to @laestrada for figuring this out!

**Comment by yantosca** [comment]
Also added `#MINVERSION` documentation to the `kpp_UserManual.tex` file in commit ec67fd33120237432a404d45815d87501241dad8.

File: .ci-pipelines/ci-testing-script.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ for this_test in $all_tests; do
     echo ""
     echo ">>>>>>>> Building the $this_test test executable <<<<<<<<<"
     echo ""
-    make -f Makefile_$this_test COMPILER=GFORTRAN
+    make -j -f Makefile_$this_test COMPILER=GFORTRAN
```
Comment (laestrada) [inline .ci-pipelines/ci-testing-script.sh:26]: What's the significance of the -j option -- is it just to parallelize?

**Review by laestrada — APPROVED** [review]
Looks good to me -- having the #MINVERSION check is a useful feature!

**Comment by yantosca** [comment]
Thanks @RolfSander.  One more feature to merge in, working on it now.

**Pull request merged**